lambdas/build-stac/utils/regex.py

```diff
@@ -31,8 +31,11 @@ def extract_dates(
     filename: str, datetime_range: events.INTERVAL
 ) -> Union[Tuple[datetime, datetime, None], Tuple[None, None, datetime]]:
     """
-    Extracts start & end or single date string from filename.
+    Extracts start & end or single date string from filename (using only the file name if the full path is provided)
     """
+    
+    filename = filename.split("/")[-1]
```

Comment:
This makes sure that we search for dates only in the file name and not in the full path.

Reply:
I believe it's only sending the filename and not the remote fileurl? 
You can verify on the failed runs of SRTMG1 😅

Reply:
Hm in my case I was ending up with the remote fileurl here 🤔...

Just looked at the failed runs and it looks unrelated to the reason I added this code -- it looks like your remote fileurl don't have any date at all.

Reply:
The error message still outputs "filename" though and it wasn't the entire s3 locations, unsure why they're different then

Reply:
Got a link to the error ?

Reply:
"errorMessage": "No dates provided in filename='N02W057.SRTMGL1.tif'. At least one date in format yyyy-mm-dd is required."

Reply:
This error emerged from a run that was using the deployment _including_ my changes here. 

My changes here add a line to make sure we look for dates only in the file name. 

After that line, if no dates are found, an error is thrown, and that's the error you got. Your file name doesn't have any dates. 

I think every thing is working as expected ?

Reply:
> I believe it's only sending the filename and not the remote fileurl?

Whatever path is in the inventory [gets passed along without being truncated](https://github.com/MAAP-Project/maap-data-pipelines/blob/18ae388d8876fb6fd33f2dd54b338d7493dc722f/lambdas/inventory/handler.py#L68C19-L68C49).

> You can verify on the failed runs of SRTMG1 😅

See my last comment

Reply:
oh right, since you deployed - I didn't know that at the time, that makes sense. 
Is there a reason that we only want to look at the last portion? if it's similar to SRTM could the date be the prefix it's stored under?

Reply:
@jjfrench 

> if it's similar to SRTM could the date be the prefix it's stored under?

When you say SRTM, are you talking about the collection corresponding to [your PR ?](https://github.com/MAAP-Project/maap-data-pipelines/pull/20) 

If so, I don't understand what you mean by 

> could the date be the prefix it's stored under

Because looking at your SRTM inventory file, I don't see any dates in the paths. That's the reason you added `start_datetime` and `end_datetime` to the step function input parameters.

Reply:
> Is there a reason that we only want to look at the last portion?

I don't remember the details, however, it was leading to incorrect parsing in the CCI case, with e.g. `s3://nasa-maap-data-store/file-staging/nasa-map/ESACCI_Biomass_L4_AGB_V4_100m_2017/N00E120_ESACCI-BIOMASS-L4-AGB-MERGED-100m-2017-fv4.0.tif`.

Reply:
Since we're not planning on using this repo for new collections, can we merge @jjfrench ?